ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,228 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	mock_session "github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/testconst"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	mock_metrics "github.com/aws/amazon-ecs-agent/ecs-agent/metrics/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	StartingLatestTaskManifestSequenceNumber = 15
+)
+
+// setupTestManifestMessage initializes a dummy TaskManifestMessage for testing
+func setupTestManifestMessage() *ecsacs.TaskManifestMessage {
+	return &ecsacs.TaskManifestMessage{
+		ClusterArn:           aws.String(testconst.ClusterName),
+		ContainerInstanceArn: aws.String(testconst.ContainerInstanceARN),
+		MessageId:            aws.String(testconst.MessageID),
+		Tasks: []*ecsacs.TaskIdentifier{
+			{DesiredStatus: aws.String("STOPPED"), TaskArn: aws.String(testconst.TaskARN)},
+		},
+		Timeline: aws.Int64(StartingLatestTaskManifestSequenceNumber),
+	}
+}
+
+// setupTestTaskManifestResponder initializes the TaskManifestResponder for testing.
+func setupTestTaskManifestResponder(t *testing.T, factory metrics.EntryFactory, f wsclient.RespondFunc) *taskManifestResponder {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Create task manifest responder.
+	mockTC := mock_session.NewMockTaskComparer(ctrl)
+	mmIDA := mock_session.NewMockManifestMessageIDAccessor(ctrl)
+	mockSNA := mock_session.NewMockSequenceNumberAccessor(ctrl)
+
+	mmIDA.EXPECT().SetMessageID("").Times(1)
+	mmIDA.SetMessageID("")
+
+	mockSNA.EXPECT().SetLatestSequenceNumber(gomock.Any()).MinTimes(1)
+	mockSNA.SetLatestSequenceNumber(setupTestManifestMessage())
+
+	testTaskManifestResponder := NewTaskManifestResponder(mockTC, mockSNA, mmIDA, factory, f)
+	return testTaskManifestResponder
+}
+
+// TestManifestAckHappyPath tests the happy path for a typical TaskManifestMessage and confirms
+// expected ack is made
+func TestManifestAckHappyPath(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	ackSent := make(chan *ecsacs.AckRequest)
+
+	testResponseSender := func(response interface{}) error {
+		resp := response.(*ecsacs.AckRequest)
+		ackSent <- resp
+		return nil
+	}
+
+	testTaskManifestResponder := setupTestTaskManifestResponder(t, mock_metrics.NewMockEntryFactory(ctrl), testResponseSender)
+
+	// After we get the latest sequence number, we will expect to set the latest
+	// message ID
+	mockmmIDA := testTaskManifestResponder.messageIDAccessor.(*mock_session.MockManifestMessageIDAccessor)
+	mockmmIDA.EXPECT().SetMessageID(gomock.Any()).Times(1)
```

Comment:
Could we be more specific than `gomock.Any()` here (i.e., use `testconst.MessageID` instead of `gomock.Any()`)? This comment also applies to anywhere else we're using `gomock.Any()` in our tests.